DE_pipeline.snake

```diff
@@ -532,7 +567,96 @@ rule DESeq2:
 		script_file = pph.log(log.out, snakemake_format(script), step="DESeq2", extension="R", contrast = "all", **wildcards)
 		shell("Rscript --vanilla '{script_file}' &>> '{log.out}'")
 		
-		
+##-------------------- scores -------------------------------------------------------------------------------
+
+rule scores:
+	"""Compute scores by ssGSEA"""
+	input:
+		counts = pph.file_path("DESeq2", "deseq2.rds", contrast = "all")
+	output:
+		scores = pph.file_path("scores", "rds", contrast = "all")
+	log:
+		out = pph.file_path("scores", "output.log", contrast = "all", log=True)
+	params:
+		genome    = config["organism"]["files"]["genome"],
+		gtf       = config["organism"]["files"]["gtf"],
+		annot_pkg = config["organism"]["R"]["annotations"],
+		gene_list = config["scores"]["path"],
+		id_class  = config["scores"]["id_class"],
+		kcdf      = config["scores"]["kcdf"],
+		min_n     = config["scores"]["min_n"]
+	run:
+		categories = "NULL"
+		if "categories" in config["scores"].keys():
+			categories = 'c("' + '", "'.join(config["scores"]["categories"]) + '")'
+
+		script = textwrap.dedent(r"""
```

Comment:
Can we consider isolating R code instead of having it in the middle of the snakemake file? How hard is it to turn this into a callable with parameters?

Reply:
I think that would be a good idea. I have already started getting rid of snakemake variables in the middle of the R code (the "no snakemake variables beyond this line" thingy). Programistically it would be a piece of cake. This is a job I could take over.